cms/djangoapps/contentstore/views/item.py

```diff
@@ -116,9 +120,9 @@ def _is_library_component_limit_reached(usage_key):
     return total_children + 1 > settings.MAX_BLOCKS_PER_CONTENT_LIBRARY
 
 
-@require_http_methods(("DELETE", "GET", "PUT", "POST", "PATCH"))
-@login_required
-@expect_json
+@api_view(["DELETE", "GET", "PUT", "POST", "PATCH"])
+@authentication_classes([JwtAuthentication, BearerAuthentication, SessionAuthentication])
```

Comment:
I think these should be:
```suggestion
@view_auth_classes()
```

Reply:
These are the decorators from django-rest-framework. The point was to adopt d-r-f methods instead of regular django views, correct?

Reply:
If you check the code of that function, you'll see that it's a helper in edx-platform that applies a consistent set of auth classes, likely to make it simpler to adapt all views together in the future.

Reply:
Oh, I see. Is it really the recommended practice, though? Not only it is mixing authentication and authorization in one single function, we are also introducing another layer of abstraction for things that are orthogonal.

Reply:
I can understand if it doesn't match your coding style or approach, but we have to match the one used in this repository. It is the idiom that is used across the repository, and I feel you should use it unless you think the use case for this API doesn't match the defaults in that helper. 

Note: I will leave it to @viadanna to decide how to take this forward since he's the reviewer here, and the reviewer for edX will be the final arbiter.

Reply:
@lullis I have to agree with @xitij2000 here, to maintain consistency with the code provided as example on https://github.com/edx/edx-platform/pull/19052